### PR TITLE
Migrate provisioning records to typed resume evidence

### PR DESCRIPTION
**Pull request by ronaldmannak**
## Summary

Refs #6; migrates provisioning records from retained draft #104 under MVP-PLAN.md §§3 and 9 and ADR 0003.

**Stacked on #165. Merge #165 first; do not merge this PR into its feature-branch base.** The reviewed storage prerequisite is included through ordinary forward integration; no history was rewritten.

- Reuse ordered stages, checkpoint/state invariants, persisted effect identities and stage-status records.
- Replace raw resume summaries/redactor coupling with three closed partial-work kinds and fixed presentation text.
- Preserve bounded relative operational staging locations byte-for-byte; refuse malformed input rather than silently changing identity. A location is neither diagnostic text nor filesystem authority.
- Use provisioning schema 2; refuse unsupported schema-1 prototypes without rewriting them. Tokenless schema-2 reservations retain their inspection-only representation.
- Keep minted counters persistable across the old decode ceiling. In-memory and persisted records now share the complete UInt64 range, with checked `nextEffectToken` exhaustion and no wrap/reset/renumbering.
- Add eighteen model/schema/evidence test functions across four suites, including twelve boundary/pending-token-promotion/exhaustion cases.

The provisioner does not run yet. Reducer/callback migration, durable StateStore integration, inspection and UI execution remain later components. A saved checkpoint is not current VM/tool readiness. The unpublished reducer is being adapted to refuse new reservations on exhaustion while preserving existing callback settlement; this model PR executes no effects.

## Composition

Head: `8e60692cccb5a18b58957f1aeb640a19d3798471`. Base: reviewed #165 at `d6d4b45fd800d31f447eca32fa1a5faf9757e554`.

**508 additions in five Core files.** The initial model migration was published at `35c55ada`; the follow-up fixes the valid [counter-persistence P2](https://github.com/PicoMLX/Guesthouse/pull/166#discussion_r3975768686). [Implementation and test rationale](https://github.com/PicoMLX/Guesthouse/pull/166#discussion_r3975837665).

## Validation

- `git diff --check` passes. The full source delta and all published counter consumers were inspected.
- Current-head Codex review completed at September 10,05:30:18 UTC with no new findings; the original PR message received the bot 👍 at05:30:21 UTC.
- **Current-head Xcode Cloud passed at September 10,05:31:44 UTC**, with zero errors, test failures, analysis issues or warnings. The accepted P2 is explained and resolved with [validation evidence](https://github.com/PicoMLX/Guesthouse/pull/166#discussion_r3975867394). All current review/check/reaction pages are complete; the PR is MERGEABLE/CLEAN.
- **Dependency hold only:** #165 is still open. Merge #165 first, then settle this PR on main and recheck composition/current-head gates. Do not merge this PR into its feature-branch base.
- Historical local validation (565 strict package functions, focused Debug/Release, CI-hook checks and unsigned Xcode build-for-testing) belongs to the original prepared migration, not this correction or #165's later regressions.
- Per the owner's instruction, current compilation/package/app tests use origin/Xcode Cloud; no fresh local package/app builds or large build caches were created.

Before merge: require exact-head green CI, matching completed review/original-message thumbs-up, explained/resolved findings, and settled parent composition after #165 lands.

## Preserved scope

#104 and #6 remain open until their complete retained scope is migrated and merged. No host mutation, process invocation, provider selection, or signed/hardware claim. Original source, tests, branches and review history remain preserved.

### CODE REVIEW

File: Packages/GuesthouseCore/Sources/GuesthouseCore/Provisioning/ProvisioningState.swift [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        guard version == Self.currentSchema else {
+            throw DecodingError.dataCorruptedError(forKey: .schemaVersion, in: container, debugDescription: "provisioning state schema \(version) is not \(Self.currentSchema)")
+        }
+        let stage = try container.decode(ProvisioningStage.self, forKey: .stage)
+        let status = try container.decode(StageStatus.self, forKey: .status)
+        guard Self.isConsistent(stage: stage, status: status) else {
+            throw DecodingError.dataCorruptedError(forKey: .status, in: container, debugDescription: "checkpoint stage does not match \(stage.rawValue)")
+        }
+        let issuedEffects = try container.decode(UInt64.self, forKey: .issuedEffects)
+        guard max(issuedEffects, status.pendingEffect?.value ?? 0) <= Self.maximumIssuedEffects else {
```
[the remark this answers is not in the document]
Comment (ronaldmannak) [reply Packages/GuesthouseCore/Sources/GuesthouseCore/Provisioning/ProvisioningState.swift]: Accepted and fixed in 8e60692cccb5a18b58957f1aeb640a19d3798471. The old decode-only half-range ceiling allowed a successful mint to produce an unreadable record. Moving that ceiling by one would only move the bug.

The in-memory and persisted models now both support the entire UInt64 range. `nextEffectToken` checks exhaustion before adding one; at UInt64.max it returns nil while the exhausted record and any outstanding token remain round-trippable. The model never wraps, resets, clamps, or renumbers an identity. Coordinators must refuse new reservations on exhaustion and preserve the existing record for recovery; the later, not-yet-published reducer migration must use this checked path.

Regression coverage now runs decode → checked next token → construction → encode → decode for five boundaries (including the old ceiling and the last issuable token), both explicit counters and pending-token promotion, plus two exhausted-record round trips. Malformed and genuinely out-of-range numbers still fail. Schema/stage checks remain intact.

`git diff --check` passed. The fix is pushed and fresh Xcode Cloud validation is running; I am not counting the previous head's green CI or historical local tests as validation of this commit. Keeping this thread open until the new head's validation completes.

**Comment by ronaldmannak** [comment]
@codex review

Please review current head `8e60692cccb5a18b58957f1aeb640a19d3798471`. The valid counter-persistence P2 is fixed with shared full-range serialization and checked exhaustion, with boundary/relaunch regression cases. Xcode Cloud is running on this head. The detailed rationale and validation status are in the original review thread; no raw logging or runtime effects were added.

File: Packages/GuesthouseCore/Sources/GuesthouseCore/Provisioning/ProvisioningState.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (ronaldmannak) [reply Packages/GuesthouseCore/Sources/GuesthouseCore/Provisioning/ProvisioningState.swift]: Validation complete for 8e60692cccb5a18b58957f1aeb640a19d3798471: Xcode Cloud passed at September 10,05:31:44 UTC with zero errors, test failures, analysis issues or warnings. The matching Codex review completed at05:30:18 UTC without new findings and the original PR message received 👍 at05:30:21 UTC. Resolving this accepted-and-fixed P2; the implementation and boundary/relaunch regression rationale are recorded above.

The fix has also been carried into the unpublished contract/reducer/recovery components so future migration cannot retain unchecked minting or the old decode ceiling. Those later source corrections are not part of this PR's CI result and require their own publication/validation. Merge #165 first, then settle this PR on main and recheck composition; do not merge into its current feature-branch base.
